### ISSUE TITLE
Add class docs

### ISSUE THREAD

**Issue by CoffeeCoder1** [samsface/godot-pure-data#14]
Add class documentation to the plugin. It looks like adding this would either require switching to SConstruct or reimplementing some of the documentation build system (see [this script](https://github.com/IvanInventor/godot-cpp/blob/cmake-rewrite/docs_generator.py)).

**Comment by samsface** [comment]
Not against moving to scons but I think moving to scons might be a ton of work. I went with CMake because libpd already had pretty good CMake support.

Another approach would be to just wrap the library in an indirection layer written in GDScript and comment up that. This has some added benefits. Quicker iteration for changing comments, no need to compile. You can validate args gdscript size. Did you know GDExtension just crashes if you pass an arg of the wrong type? Can also fill it would more could helper functions and stuff in gdscript which is easier to maintain compared to c++.

**Comment by CoffeeCoder1** [comment]
> Not against moving to scons but I think moving to scons might be a ton of work.

I totally agree with that.

> Another approach would be to just wrap the library in an indirection layer written in GDScript and comment up that.

I wouldn't be against this, but it may prove to be a bit difficult for some things. As far as I can tell, the APIs that you need to make native Godot classes have GDScript bindings but are only documented in C++, so it would probably be a pain to get it working (these appear to be needed for things like import plugins to work properly), although we could always just make classes that need to be native in C++ (for example, a resource class to hold the data in a patch file).

**Comment by CoffeeCoder1** [comment]
I agree that it would make it a lot easier to do it this way, I just have had problems in some specific scenarios because classes created with `class_name` behave a bit differently than types created in C++ (the one example that comes to mind is import plugins, which needed [this workaround](https://github.com/samsface/godot-pure-data/blob/b52d2f321c4032d60a429ac5e8df2c332a323700/project/addons/pure-data/pure_data.gd#L8) to get working and still don't seem to work properly - the option to reimport is gone and the files don't show up in the quick load menu, while both of those work if the resource type is set to a native type). I do think, where possible, it isn't a bad idea to create GDScript abstractions, we just have to keep in mind that it might not work everywhere (I believe this is caused because native types are "real" types while GDScript classes are just an object of the type that they inherit with a script attached).

**Comment by CoffeeCoder1** [comment]
I have found another pretty similar library ([fediazc/godot-pd](https://github.com/fediazc/godot-pd)) which uses SCons to build instead of CMake that we can use as a reference for this library if we end up switching to SCons.

**Comment by CoffeeCoder1** [comment]
With version 4.4, Godot has proper documentation display for GDScript named classes ([docs](https://docs.godotengine.org/en/stable/tutorials/scripting/gdscript/gdscript_documentation_comments.html)), so if we just wrap everything in GDScript that could handle all the class docs (although I believe imported resource types still have to be made in C++, so that just might not be able to have class docs).
